[Luo Tian You] iP

HakkaNgin opened the pull request.

Commit:
Add Gradle support

HakkaNgin committed:
Level 0: Greet

HakkaNgin committed:
Level 1: Bot greets user, echos input and says bye to user if input is bye

HakkaNgin committed:
Level 1: Fixed minor bug. Should not echo if input is bye

HakkaNgin committed:
Level 1: Made fields and methods in class Pixel private

HakkaNgin committed:
Level 2: Created list, added the ability to store whatever text entered by the user and display them back to the user when requested.

HakkaNgin committed:
Level 3: Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

HakkaNgin committed:
Level 4:  ToDos, Events, Deadlines

HakkaNgin committed:
Level 4: Correctly implemented todo, event and deadline

HakkaNgin committed:
A-TextUiTesting: Fixed minor bug, label display for Deadline, ToDo and Event (D/T/E)

HakkaNgin committed:
A-TextUiTesting: Finalised tests, fixed the bug where command text (todo/ event/ deadline) is wrongly displayed along with the actual message. Also accounted for the case where the user does not set a due date by using /

HakkaNgin committed:
Level-5: Added try-catch blocks

HakkaNgin committed:
Level-6: Switched from array to ArrayList to store the Tasks. Also implemented deletion of tasks. But code is still quite buggy. Also level of abstraction of code is too low and needs to be further refined.

HakkaNgin committed:
Level-5.1: Created custom exception for incorrect format of input (not a task or command)

HakkaNgin committed:
Level-5.2: Created greater abstraction of the methods when handling the different types of tasks. No longer accepts input which are not tasks or commands. Still buggy as I omitted the /at command for user input.

HakkaNgin committed:
Level-5.3: Fixed bugs. Allow both by: and at: to be displayed in the description of the task after the user types /by and /at respectively

HakkaNgin committed:
Level-6: Fixed bugs for the delete function. Allowed an element in the list (ArrayList) to be deleted and all the subsequent elements to be pushed forward by 1

HakkaNgin committed:
Level-6: Allows programme to exit after user types bye

HakkaNgin committed:
For the pull request

HakkaNgin committed:
Level-7: Allows users to save tasks in the dard disk. WARNING: Users are not able to read from the file and changes made to the tasks in the programme will not be reflected in the file

HakkaNgin committed:
Level-8: Added feature that checks whether user input is in yyyy-MM-dd HHmm format, and subsequently formats date and time in <name of month> dd yyyy hh:mm aa format

HakkaNgin committed:
Level-8.1: Somehow my git still says that some changes are not committed

HakkaNgin committed:
Merge pull request #1 from HakkaNgin/branch-Level-7

Level-7: Allows users to save tasks in the dard disk. WARNING: Users …

HakkaNgin committed:
Merge branch 'master' into branch-Level-8

HakkaNgin committed:
Merge pull request #2 from HakkaNgin/branch-Level-8

Branch level 8

HakkaNgin committed:
Level-8.2: Resolved conflicts manually on IDE (on main branch) before merging level 8 branch. Also made a new pull request on github with my previous level 8 commit and resolved the merge conflict manually on Github as well. Not very sure if that's the correct procedure -- to handle conflicts twice, on both Github and the IDE

HakkaNgin committed:
Merge branch 'master' of https://github.com/HakkaNgin/ip

HakkaNgin committed:
A-MoreOOP: defined four new classes -- Ui for user interface, Storage to handle writing to text, Parser (nested class) to deal with interpreting the user command, TaskList to handle tasks in the list

HakkaNgin committed:
Level 7.1: Fixed bug where new tasks will override older tasks on pixel.txt. Also converted to BufferedWriter for more effective writing to file.

HakkaNgin committed:
Bug repair 1: Fixed bug in conditional statement on length of list in Pixel.java

HakkaNgin committed:
Level-7.2: Fixed bugs. Now all edits will be updated to pixel.txt

HakkaNgin committed:
A-Packages: Organised files into packages

HakkaNgin committed:
Merge remote-tracking branch '2103main/add-gradle-support'

HakkaNgin committed:
A-Gradle: Set up Gradle to run unit tests

HakkaNgin committed:
Bug repair 2: When the user does not indicate a due date, (by: ) and (at: ) will not be displayed

HakkaNgin committed:
A-JUnit: Added JUnit tests. Also abstracted out the delete task method to the Storage class.

HakkaNgin committed:
A-JavaDoc

HakkaNgin committed:
A-CodingStandard

HakkaNgin committed:
Merge pull request #3 from HakkaNgin/branch-A-CodingStandard

A-CodingStandard

HakkaNgin committed:
Revert "A-JUnit: Added JUnit tests. Also abstracted out the delete task method to the Storage class."

This reverts commit 470508d41910526f5f0c03f8ac96a5a760fc817b.

HakkaNgin committed:
branch-A-JavaDoc: Added JavaDocs to some classes and methods

HakkaNgin committed:
Revert "A-JavaDoc"
Revert as I need to redo my branchings
This reverts commit 3fdd2e6617997cffc7c68eff4159acbcc7d16e4a.

HakkaNgin committed:
branch-A-CodingStandard: Re-do as my submission yesterday did not comply with the branching instructions

HakkaNgin committed:
Level-9 (redo): Added feature for users to find all tasks which description matches the query string

HakkaNgin committed:
Merge branch 'master' of https://github.com/HakkaNgin/ip

HakkaNgin committed:
Week 3: Master branch redo -- because I forgot to branch out when making the previous changes. Had to revert all changes

HakkaNgin committed:
Merge branch 'master' into branch-A-JavaDoc

HakkaNgin committed:
Merge pull request #4 from HakkaNgin/branch-A-JavaDoc

Merge A-JavaDoc with master branch

HakkaNgin committed:
Merge branch 'master' into branch-A-CodingStandard

HakkaNgin committed:
Merge pull request #5 from HakkaNgin/branch-A-CodingStandard

Re-merge branch-A-CodingStandard with master

HakkaNgin committed:
Merge pull request #6 from HakkaNgin/branch-Level-9

Level-9 (redo): Added feature for users to find all tasks which descr…

HakkaNgin committed:
Merge pull request #7 from HakkaNgin/master

Master --> branch-A-JavaDoc

HakkaNgin committed:
A-JavaDocs: Trying this again as somehow my JavaDocs don't get merged into master

HakkaNgin committed:
A-JavaDocs: Trying this again as somehow my JavaDocs don't get merged into master

HakkaNgin committed:
Merge pull request #8 from HakkaNgin/branch-A-JavaDoc

Merging JavaDocs once and for all (

HakkaNgin committed:
Final commit for week 3

HakkaNgin committed:
Bug Fix: Making the programme more OOP-Oriented

HakkaNgin committed:
Update README.md

SeekSaveServe commented on src/main/java/pixel/Pixel.java:
```diff
@@ -0,0 +1,38 @@
+    private final Scanner myScanner = new Scanner(System.in);  // Create a Scanner object
+
+    // the following fields are made final to facilitate testing, should be private
+    // private final String filePath;
+    public final Parser parser;
+
+    public Pixel(String filePath) {
+        // this.filePath = filePath;
```
Should the commented out code be removed?

SeekSaveServe commented on src/main/java/pixel/task/Task.java:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * converts input date&time from yyyy-MM-dd HHmm to MONTH dd yyyy hh:mm aa format
```
Should the doc begin in capital letters?

SeekSaveServe commented on src/main/java/pixel/util/Parser.java:
```diff
@@ -0,0 +1,142 @@
+                int indexToChange = Character.getNumericValue(temp.charAt(0));
+                // System.out.println(indexToChange);
+                if ((indexToChange > 0) && (indexToChange < 100)) {
+                    Storage.INPUT_TASKS.get(indexToChange - 1).markAsNotDone();
+                }
+
+                Storage.resetFile(this.filePath);
+                for (Task task : Storage.INPUT_TASKS) {
+                    Storage.appendToFile(task, this.filePath);
+                }
+
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(Storage.INPUT_TASKS.get(indexToChange - 1));
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().equals("list")) {
+                // System.out.println(inputMemory.length);
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < Pixel.count; i++) {
+                    Task currentTask = Storage.INPUT_TASKS.get(i);
+                    System.out.println((i + 1) + ". " + currentTask);
+                }
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().startsWith("delete ")) {
+                Storage.deleteEntry(userInput, filePath);
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().startsWith("find ")) {
+                ArrayList<Task> results = Storage.findEntry(userInput);
+                System.out.println("Here are the matching tasks in your list:");
+                for (int i = 0; i < results.size(); i++) {
+                    Task currentTask = results.get(i);
+                    System.out.println((i + 1) + ". " + currentTask);
+                }
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else {
+                throw new IncorrectFormatException("Input should be a task or command!"); // programme breaks
+            }
```
Is switch better here?

xhphoong replied:
I like that you separated by one blank line for each logical units within a block, make it easier to read.

SeekSaveServe commented on src/main/java/pixel/util/Storage.java:
```diff
@@ -0,0 +1,128 @@
+        return resultTasks;
+    }
+
+    /* During the write operation, the characters are written to the internal buffer instead of the disk.
+    Once the buffer is filled or the writer is closed, the whole characters in the buffer are
+    written to the disk.
+    Hence, the number of communication to the disk is reduced.
+    */
```
Should this be removed?

SeekSaveServe reviewed (approved):
Overall, great commit!

xhphoong commented on src/main/java/pixel/Pixel.java:
```diff
@@ -0,0 +1,38 @@
+
+import pixel.util.*;
+
+import java.util.Scanner;
```
I like that you listed out explicitly for the imported java.util package. 
But perhaps the pixel.util should also follow the same way as the java.util (explicitly listing out the imported class instead of using "*") ?

xhphoong commented on src/main/java/pixel/task/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+ */
+public class Deadline extends Task {
+
+    public static final String TAG = "D";
```
I like that you adhere to the naming convention for constant.

xhphoong commented on src/main/java/pixel/task/Event.java:
```diff
@@ -0,0 +1,37 @@
+package pixel.task;
```
Good job in grouping the class in the respective package.

xhphoong commented on src/main/java/pixel/task/Task.java:
```diff
@@ -0,0 +1,166 @@
+package pixel.task;
+
+import pixel.util.DateValidator;
+import pixel.util.UserInterface;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+import java.util.Date;
+import java.util.Locale;
```
I like your consistent ordering of import statements,  easier for me to read.

xhphoong reviewed (commented):
Overall, neat code and easy to read but it would be even better for readability if the commented out code is removed.

HakkaNgin committed:
Update README.md

HakkaNgin committed:
Level 1: Implemented a scaffold for GUI

HakkaNgin committed:
Level 10: Implemented a scaffold for GUI

HakkaNgin committed:
Update README.md

HakkaNgin committed:
Minor fix: Got rid of dead code

HakkaNgin committed:
Merge branch 'master' of https://github.com/HakkaNgin/ip

HakkaNgin committed:
CheckStyle: Use checkStyle to detect coding style violations.

HakkaNgin committed:
Merge pull request #10 from HakkaNgin/branch-A-CheckStyle

A-CheckStyle: Use CheckStyle to detect coding style violations.

HakkaNgin committed:
A-FullCommitMessage

This is a test commit message.

Attempted to make the GUI display
a welcome message upon initialisation. (currently unsuccessful)

A welcome message is more user-friendly and enables users to view
the set of instructions before making any command

Let's try to make this feature successful

HakkaNgin committed:
Added assertion statements at parts of the code
where an element is added or removed from the ArrayList
to check if the final size of the ArrayList is correct
Also checked whether the tags of each task is correct

HakkaNgin committed:
Merge pull request #11 from HakkaNgin/branch-A-Assertions

Added assertion statements at parts of the code

HakkaNgin committed:
A-Enums: Added Enumerations to handle the mark and unmark cases

HakkaNgin committed:
A-CodeQuality: Refractored the code, mainly in the Parser class. Main objective is to result in greater abstraction

HakkaNgin committed:
Merge pull request #12 from HakkaNgin/branch-A-CodeQuality

Merge A-CodeQuality (containing A-Enums) into master

HakkaNgin committed:
Added DuplicateEntryException to handle duplicate entries

HakkaNgin committed:
Resolved some branching conflicts

HakkaNgin committed:
C-Detect Duplicates: Add the ability to recognize and deal with duplicate items. e.g., the same task added multiple times.

HakkaNgin committed:
Merge pull request #13 from HakkaNgin/branch-C-DetectDuplicates

Extension Feature: Branch C -- Detect Duplicates

HakkaNgin committed:
Git Flavoured Markdown: Forgot to include in new tag in week 4 submission

HakkaNgin committed:
Included welcome message in GUI

HakkaNgin committed:
Added new feature in Storage to read Task data from text file

HakkaNgin committed:
A-MoreErrorHandling
Added more format checking for external text file to store tasks.
Creates new file if file does not exits
Throws exception when tasks recorded in file are in invalid format

HakkaNgin committed:
Optimised some aspects of error handling of invalid text file formatting

HakkaNgin committed:
Fixes bug from parsing user input of date and time

HakkaNgin committed:
More testing -- added more tests cases
to test the functionality of Pixel in different formats
of date and time inputs of tasks

HakkaNgin committed:
Finalised JavaDocs and Checkstyle for all .java files

HakkaNgin committed:
Made some minor changes before final IP submission

HakkaNgin committed:
Update README.md

HakkaNgin committed:
A-UserGuide

HakkaNgin committed:
Merge branch 'master' of https://github.com/HakkaNgin/ip

HakkaNgin committed:
Add files via upload

HakkaNgin committed:
Update README.md

HakkaNgin committed:
Merge pull request #15 from HakkaNgin/A-Release

Embedded Ui.png into user guide

HakkaNgin committed:
Previously uncommitted changes -- bug fix for Jar file

HakkaNgin committed:
Merge branch 'master' of https://github.com/HakkaNgin/ip

HakkaNgin committed:
Delete Ui.png

HakkaNgin committed:
Add files via upload

HakkaNgin committed:
Update README.md
